apps/www/src/routes/portal/admin/+page.server.ts

```diff
@@ -0,0 +1,37 @@
+import { UserService } from '$lib/services/user.service';
+import { redirect } from '@sveltejs/kit';
+
+export async function load({ locals }) {
+	if (!locals.user || locals.user.role !== 'board') {
+		throw redirect(303, '/portal');
+	}
+
+	const userService = new UserService(locals.db);
+	const users = await userService.findAll();
+
+	return {
+		user: locals.user,
+		users
+	};
+}
+
+export const actions = {
+	updateRole: async ({ request, locals }) => {
+		const formData = await request.formData();
+		const userId = formData.get('userId');
+		const role = formData.get('role');
+
+		if (!locals.user || locals.user.role !== 'board') {
+			return { status: 401, body: { error: 'Unauthorized' } };
```

Comment:
Når en form action failer bruker main `fail` fra `@sveltejs/kit`

```suggestion
			return fail(401, { error: 'Unauthorized' });
```